swap/swap_test.go

```diff
@@ -1009,31 +1026,37 @@ func TestValidateCode(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Error in deploy: %v", err)
 	}
-
 	testBackend.Commit()
 
-	if err = cswap.ValidateCode(context.TODO(), testBackend, swap.GetParams().ContractAddress); err != nil {
+	factory, err := cswap.FactoryAt(testBackend.factoryAddress, testBackend)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if err = factory.VerifyContract(swap.GetParams().ContractAddress); err != nil {
 		t.Fatalf("Contract verification failed: %v", err)
 	}
 }
 
-// tests if ValidateCode rejects an address with different bytecode
-func TestValidateWrongCode(t *testing.T) {
+// tests if VerifyContract rejects a chequebook not deployed by that factory
+func TestVerifyContractNotDeployedByFactory(t *testing.T) {
 	swap, clean := newTestSwap(t, ownerKey)
 	defer clean()
 
 	opts := bind.NewKeyedTransactor(ownerKey)
 
-	// we deploy the ECDSA library of OpenZeppelin which has a different bytecode than swap
-	addr, _, _, err := contract.DeployECDSA(opts, swap.backend)
+	addr, _, _, err := contractFactory.DeploySimpleSwap(opts, swap.backend, ownerAddress, big.NewInt(int64(defaultHarddepositTimeoutDuration)))
 	if err != nil {
 		t.Fatalf("Error in deploy: %v", err)
 	}
-
 	testBackend.Commit()
 
-	// since the bytecode is different this should throw an error
-	if err = cswap.ValidateCode(context.TODO(), swap.backend, addr); err != cswap.ErrNotASwapContract {
+	factory, err := cswap.FactoryAt(testBackend.factoryAddress, testBackend)
```

Comment:
No need to create a factory here, it is already part of swap

Reply:
Correct, this was a remnant from when startChequebook and new were called at different times.